### ISSUE TITLE
App fails to load https://music.youtube.com with error

### ISSUE THREAD

**Issue by jscheel** [steve228uk/YouTube-Music#163]
I've started getting a blank screen when opening the app. I built from source, enabled the inspector, and I'm seeing this error:

```
[Error] FetchEvent.respondWith received an error: TypeError: undefined is not an object (evaluating 'a.preloadResponse.then')
[Error] Cannot load https://music.youtube.com/.
[Error] Failed to load resource: FetchEvent.respondWith received an error: TypeError: undefined is not an object (evaluating 'a.preloadResponse.then') (music.youtube.com, line 0)
```

I can get it to go to any other page, including Youtube itself. But Youtube Music fails every time now.

**Comment by jscheel** [comment]
Also getting this from the app's debugging:

```
[Process] 0x7fc67680b220 - [pageProxyID=5, webPageID=6, PID=7102] WebPageProxy::didFailProvisionalLoadForFrame: frameID=4, domain=WebKitServiceWorker, code=0
```

**Comment by jscheel** [comment]
So, super weird, but I ended up just clearing all the data out before loading the web view via: 

```
WKWebsiteDataStore.default().removeData(ofTypes: WKWebsiteDataStore.allWebsiteDataTypes(), modifiedSince: Date(timeIntervalSince1970: 0), completionHandler: {})
```

and it worked. Bonkers.

**Comment by evsasse** [comment]
> and it loaded. However, nothing is playing.

Similar stuff happening to me. It was working fine until yesterday. But since this morning nothing is playing; while navigation/search/etc works as usual.

**Comment by juampapo546** [comment]
I'm having a similar issue, just downloaded from release tags and nothing is playing. i'm using big sur

**Comment by TimOliver** [comment]
Hrmmm. I also periodically open the app to a blank screen. Usually if I quit and re-open the app, it'll eventually start working again.

Thanks for the debug messages! I hadn't managed to capture any on my end. From those messages, do you think it's something on our end due to the custom JS we inject?

**Comment by jscheel** [comment]
@TimOliver no, unfortunately it's not. This is caused by a bad service worker registration. The error that is throw is coming from the service worker trying to handle a prefetch of the file, and crashing because the fetch event doesn't support preloadResponse. Makes me think there's something wrong with the cache maybe. No clue how it could have possibly gotten into that state. However, I ended up unregistering the service worker and got it working again. Unfortunately, on the next load, it's broken again.

I found a bug report that looks like the issue here: https://developer.apple.com/forums/thread/702755. I tried the suggested solution (loading a different page first), but it seems to only work intermittently.

**Comment by jscheel** [comment]
Ok, I have a possible solution. It's a hammer, but it seems to be working: I've set up a ContentRuleList in the WkWebView to prevent `sw.js` from loading. This prevents the service worker from registering, which in turn prevents the prefetching/caching of the code that seems to be broken for some reason. I need to clean up all my other test code to make sure this works consistently.

**Comment by mark-pictor-csec** [comment]
Not sure if it's related, but I just allowed an update to the latest version and now it won't play. Could be it's unrelated and it was already broken for me - this is the first time today I've tried it.

**Comment by TimOliver** [comment]
Thanks @jscheel! I've merged your PR and released a new update. It should download the next time you open the app.

@mark-pictor-csec Try opening the app and downloading the latest update, and see if the problem persists for you after that. :)

**Comment by evsasse** [comment]
No luck for me, same problem still happening. Updated it, uninstalled/reinstalled using brew, same behavior as before...

I see that someone created another similar issue, I will keep an eye there https://github.com/steve228uk/YouTube-Music/issues/165

**Comment by TimOliver** [comment]
@evsasse Uh. Whoops. Try downloading the release directly from GitHub and give that one a try. I haven't implemented automated releases to Brew yet. 😅

**Comment by evsasse** [comment]
> I haven't implemented automated releases to Brew yet. 😅

But the "Check for updates" works, right? I am able to get 1.3.2, even though I installed with brew originally 👍.

>  Try downloading the release directly from GitHub and give that one a try.

Uninstalled from brew, and tried that just now... Still no luck :/

Navigation/search/etc works fine. But when I click to start playing something, it keeps loading forever, doesnt show the "bar" at the bottom.

<img width="1049" alt="Screen Shot 2022-05-03 at 10 28 23" src="https://user-images.githubusercontent.com/7394653/166461730-58deff7f-e97c-4011-9d37-909f4a9a9dd1.png">

**Comment by jscheel** [comment]
I’m thinking we might have to add code to force-unregister the service
workers that are already installed for people. I’ll make an update for that
asap.

**Comment by jscheel** [comment]
Ok @TimOliver, I've added a new PR for unregistering service workers

**Comment by evsasse** [comment]
1.3.3 seems to have fixed it for me. Thanks @jscheel @TimOliver 🙇!

**Comment by Raheemshah2809** [comment]
Should Close this no, as I don't see this anymore as well

**Comment by targyll** [comment]
I have just discovered this app (v1.3.3). It is not working here (10.13.6), I have a blank screen and nothing more...
